src/tests/utils/PlaygroundApp.test.js

```diff
@@ -0,0 +1,51 @@
+import { playgroundApp } from '../../utils/PlaygroundApp';
+import * as processing from '../../utils/Processing';
+import Patient from './fixtures/StructureDefinition-Patient.json';
+import StructureDefinition from './fixtures/StructureDefinition-StructureDefinition.json';
+import 'fake-indexeddb/auto';
+import { FHIRDefinitions } from 'fsh-sushi/dist/fhirdefs';
+
+describe('#playgroundApp', () => {
+  it('should return an undefined package when we get invalid FHIRDefinitions', async () => {
+    const FHIRDefs = new FHIRDefinitions();
+    const loadDefsSpy = jest
+      .spyOn(processing, 'loadExternalDependenciesPlayground')
+      .mockReset()
+      .mockResolvedValue(FHIRDefs);
+    const text =
+      'Profile: FishPatient Parent: Patient Id: fish-patient Title: "Fish Patient" Description: "A patient that is a type of fish."';
+    const outPackage = await playgroundApp(text);
+    expect(loadDefsSpy).toHaveBeenCalled();
+    expect(outPackage).toBeUndefined();
+  });
+
+  it('should return the correct output pacakge when proper FSH code is entered', async () => {
+    const FHIRDefs = new FHIRDefinitions();
+    FHIRDefs.add(Patient);
+    FHIRDefs.add(StructureDefinition);
+    const loadSpy = jest
+      .spyOn(processing, 'loadExternalDependenciesPlayground')
+      .mockReset()
+      .mockResolvedValue(FHIRDefs);
+    const text =
+      'Profile: FishPatient\nParent: Patient\nId: fish-patient\nTitle: "Fish Patient"\n Description: "A patient that is a type of fish."';
+    const outPackage = await playgroundApp(text);
+    expect(loadSpy).toHaveBeenCalled();
+    expect(outPackage.profiles).toHaveLength(1);
+  });
+
+  it('should return an undefined package when the config is invalid', async () => {
+    const configSpy = jest.spyOn(processing, 'readConfigPlayground').mockImplementation(() => {
+      throw new Error('Bad Config');
+    });
+    const loadSpyConfig = jest
+      .spyOn(processing, 'loadExternalDependenciesPlayground')
+      .mockReset()
+      .mockResolvedValue(new FHIRDefinitions());
+
+    const outPackage = await playgroundApp();
+    expect(configSpy).toHaveBeenCalled();
+    expect(loadSpyConfig).toHaveBeenCalledTimes(0);
+    expect(outPackage).toBeUndefined();
+  });
```

Comment:
If this test is really difficult, we can probably live without it for now. I think the only case you don't have a test here for is the case if the `fillTank` throws an error. I think maybe we should have a test similar to the one for the config that checks the `exportFHIR` function is never called and we return undefined if `fillTank` throws an error.

Reply:
Looks good! Thanks for adding it!